src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -1135,6 +1135,13 @@ ss::future<cluster::init_tm_tx_reply> tx_gateway_frontend::limit_init_tm_tx(
     }
     auto term = term_opt.value();
 
+    auto units = co_await stm->lock_tx(tx_id, "init_tm_tx");
```

Comment:
can you add more details here why is this commit making evicting transactions less disruptive ?